go/libraries/doltcore/merge/mutable_secondary_index.go

```diff
@@ -140,12 +142,18 @@ func (m MutableSecondaryIdx) InsertEntry(ctx context.Context, key, newValue val.
 // UpdateEntry modifies the corresponding secondary index entry given the key
 // and curr/new values of the primary row.
 func (m MutableSecondaryIdx) UpdateEntry(ctx context.Context, key, currValue, newValue val.Tuple) error {
-	currKey, err := m.builder.SecondaryKeyFromRow(ctx, key, currValue)
+	currKey, err := m.leftBuilder.SecondaryKeyFromRow(ctx, key, currValue)
 	if err != nil {
 		return err
 	}
 
-	newKey, err := m.builder.SecondaryKeyFromRow(ctx, key, newValue)
+	defer func() {
+		if r := recover(); r != nil {
+			currKey, err = m.leftBuilder.SecondaryKeyFromRow(ctx, key, currValue)
+			_ = currKey
+		}
+	}()
```

Comment:
Is this needed or was this just for debugging?

Reply:
Whoops. That was for debugging. Removed.